File: TrackMe-Main/frontend/src/app/core/services/alert.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import {Alert} from '../../shared/models/alert.model';


@Injectable({
  providedIn: 'root'
})
export class AlertService {
  private apiUrl = 'http://localhost:8080/api/alerts';

  constructor(private http: HttpClient) {}

  getAllAlerts(): Observable<Alert[]> {
    return this.http.get<Alert[]>(this.apiUrl);
  }

  getAlertsByStatus(status: string): Observable<Alert[]> {
    return this.http.get<Alert[]>(`${this.apiUrl}/status/${status}`);
  }

  getAlertsByType(type: string): Observable<Alert[]> {
    return this.http.get<Alert[]>(`${this.apiUrl}/type/${type}`);
  }

  getAlertsByBraceletId(braceletId: number): Observable<Alert[]> {
    return this.http.get<Alert[]>(`${this.apiUrl}/bracelet/${braceletId}`);
  }
}

File: TrackMe-Main/backend/src/main/java/org/youcode/trackme/services/AlertService.java
package org.youcode.trackme.services;

import org.youcode.trackme.dtos.alert.AlertDTO;

import java.util.List;

public interface AlertService {
    List<AlertDTO> getAllAlerts();
    List<AlertDTO> getAlertsByStatus(String status);
    List<AlertDTO> getAlertsByType(String type);
    List<AlertDTO> getAlertsByBraceletId(Long braceletId);
}
